src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixExtension.groovy

```diff
@@ -45,15 +46,24 @@ class ScalafixExtension {
 
     ScalafixExtension(Project project) {
         this.project = project
-        configFile = project.objects.fileProperty().convention(
-                locateConfigFile(project) ?: locateConfigFile(project.rootProject))
+        final boolean isGradle4 = Gradle.isVersion4(project)
+        configFile = isGradle4? project.layout.fileProperty(): project.objects.fileProperty()
+
+        RegularFile defaultRegularFile = locateConfigFile(project)?: locateConfigFile(project.rootProject)
+        if (defaultRegularFile) {
+            if (isGradle4) {
+                configFile.set(defaultRegularFile)
+            } else {
+                configFile.convention(defaultRegularFile)
+            }
+        }
```

Comment:
I wonder if it wouldn't be cleaner to this a bit different:

```scala
class GradleCompat {
    private static boolean isVersion4(Project project) {
        project.gradle.gradleVersion.startsWith("4.")
    }

    static RegularFileProperty  fileProperty(Project project, RegularFile defaultFile = null) {
      def fileProp = isVersion4 ? project.layout.fileProperty(): project.objects.fileProperty()
      if (defaultFile) { 
            if (isGradle4) {
                fileProp.set(defaultRegularFile)
            } else {
                fileProp.convention(defaultRegularFile)
            }
       }
      fileProp
    }
}
```

then you can reuse it in both places and avoid the IFs

Reply:
Done! Thanks for the suggestion!